src/bin/cargo.rs

```diff
@@ -309,7 +311,7 @@ fn execute_external_subcommand(config: &Config, cmd: &str, args: &[String]) -> C
     let command = match path {
         Some(command) => command,
         None => {
-            return Err(human(match find_closest(config, cmd) {
+            return Err(CargoError::from(match find_closest(config, cmd) {
```

Comment:
Where possible I tend to find `format!(...).into()` a little more ergonomic than wrapping with `CargoError::from`, but that's not always possible due to inference issues.